.github/workflows/ghstack_rebase.yaml

```diff
@@ -0,0 +1,110 @@
+# The rebase-command workflow runs rebase + ghstack submit on the PR
+#
+# This workflow is triggered by leaving a "/rebase" comment on
+# a pull request. If the required preconditions are met, it will
+# rebase a stack of PRs that includes the current one as well as
+# all the PRs underneath it in the stack.
+#
+# Conditions are checked through the "ghstack-perm-check.py"
+# python script defined in the .github/workflows/scripts folder.
+#
+# Condition for the "/rebase" command are:
+# - the current PR is in the format generated by ghstack
+# - either the issuer is the author of the PR
+# - or the issuer is a maintainer
+#
+# The PRs are rebased by a bot account "fms-cibot" which has push
+# permissions on the repo.
+#
+# When a rebase is executed successfully, the horray emoji is added
+# When a rebase execution fails, a comment is added with a link to the logs
+
+name: Rebase ghstack PRs
+on:
+  repository_dispatch:
+    types: [rebase-command]
+
+jobs:
+  ghstack_rebase:
+    name: ghstack Rebase
+    runs-on: ubuntu-latest
+    steps:
+    - name: Show Environment Variables
+      run: env
+    - name: Show Github Object
+      run: |
+        cat <<'EOF'
+        ${{ toJson(github) }}
+        EOF
+    - name: Show Github Event Path Json
+      run: 'cat "$GITHUB_EVENT_PATH" || true'
+    - uses: actions/checkout@v4
+      with:
+        token: ${{ secrets.LAND_TOKEN }}
+        fetch-depth: '0'
+    - uses: actions/setup-python@v4
+      with:
+        python-version: 3.9
+    - name: Check Rebase Preconditions
+      run: |
+        pip install requests ghstack
+        .github/workflows/scripts/ghstack-perm-check.py rebase <<'EOF'
+        ${{ toJson(github) }}
+        EOF
+      env:
+        GITHUB_TOKEN: ${{ secrets.LAND_TOKEN }}
+    - name: Rebase It!
+      id: rebaseStack
+      run: |
+        # Configure Git and GHStack
+        # TODO: update the bot e-mail once a correct e-mail is available
+        git config --global user.email "foundation-model-stack@googlegroups.com"
+        git config --global user.name "Foundation Model Stack CI Bot"
+
+        cat <<EOF > ~/.ghstackrc
+        [ghstack]
+        github_url = github.com
+        github_oauth = $GITHUB_TOKEN
+        github_username = afrittoli
+        remote_name = origin
+        EOF
+
+        # Checkout the */orig committer branch for rebase 
+        # See https://github.com/ezyang/ghstack?tab=readme-ov-file#structure-of-submitted-pull-requests
+        HEAD_REF=$(echo '${{ toJson(github) }}' | jq -r .event.client_payload.pull_request.head.ref)
+        ORIG_REF=$(printf "$HEAD_REF" | sed 's,/head$,/orig,g')
+        git checkout -b rebase-branch "origin/${ORIG_REF}"
```

Comment:
can this just be `ghstack checkout <PR URL>` ?

Reply:
Yes, I think so. 

I guess `ghstack` will make GitHub API calls to determine what branches that specific PR URL corresponds to, and I was trying to avoid that, since we have all the info locally, we only need to checkout the branch.

Happy to switch to the `checkout` option if you prefer so.

Reply:
hm... I think the ghstack command will be easier to understand and maintain. I guess we could keep this with a comment explaining the equivalence though if we make it clear that this particular three lines is just the same as `ghstack checkout <pr>` without a network request